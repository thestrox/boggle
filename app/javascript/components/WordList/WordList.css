.wordlist {
    border: 1px solid lightgray;
    padding-top: 0 !important;
    padding-bottom: 0 !important;
    overflow: auto;
}

.wordlist-item {
    justify-content: space-between !important;
}