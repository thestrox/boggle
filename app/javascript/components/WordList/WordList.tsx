import React from "react";

export type WordListProp = {
    wordScoreMap: { [key: string]: number }
}

export default function WordList({ wordScoreMap }: WordListProp) {

    return (
        <ul>
            {Object.keys(wordScoreMap).map(word => {
                return (
                    <li key={word}>{word} : {wordScoreMap[word]}</li>
                );
            })}
        </ul>
    )
}