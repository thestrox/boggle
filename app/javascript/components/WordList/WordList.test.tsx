import WordList from "./WordList";
import React from 'react';
import { shallow } from 'enzyme';
import { List, ListItem } from "@material-ui/core";

function setup() {
    const props = {
        wordScoreMap: {abc: 3},
        className: 'test-class'
    };

    const component = shallow(<WordList {...props}/>);
    return {
        props,
        component
    };
}

describe('WordList Component', () => {
    it('should render correctly with given props', () => {
        const {component, props} = setup();
        const listNode = component.find(List);
        expect(listNode.hasClass('wordlist')).toBe(true);
        expect(listNode.hasClass(props.className)).toBe(true);
        
        expect(component.find(ListItem).hasClass('wordlist-item')).toBe(true);

        const spanNodes = component.find('span');
        expect(spanNodes.at(0).text()).toBe(Object.keys(props.wordScoreMap)[0] + ':');
        expect(spanNodes.at(1).text()).toBe(props.wordScoreMap.abc.toString());

    });

    it('should test snapshot renders', () => {
        const {component} = setup();
        expect(component).toMatchSnapshot();
    });
})
