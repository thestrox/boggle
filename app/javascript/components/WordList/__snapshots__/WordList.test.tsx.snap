// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList Component should test snapshot renders 1`] = `
<WithStyles(ForwardRef(List))
  className="wordlist test-class"
>
  <WithStyles(ForwardRef(ListItem))
    className="wordlist-item"
  >
    <span>
      abc
      :
    </span>
    <span>
      3
    </span>
  </WithStyles(ForwardRef(ListItem))>
  <WithStyles(ForwardRef(Divider)) />
</WithStyles(ForwardRef(List))>
`;
